src/rdkafka_cgrp.c

```diff
@@ -1189,8 +1667,18 @@ static void rd_kafka_cgrp_handle_JoinGroup (rd_kafka_t *rk,
                         rd_interval_reset(&rkcg->rkcg_join_intvl);
                 }
 
-                rd_kafka_cgrp_set_join_state(rkcg,
-                                             RD_KAFKA_CGRP_JOIN_STATE_INIT);
+                if (rd_kafka_cgrp_rebalance_protocol(rkcg) ==
+                    RD_KAFKA_REBALANCE_PROTOCOL_COOPERATIVE &&
+                    (ErrorCode == RD_KAFKA_RESP_ERR_ILLEGAL_GENERATION ||
+                     ErrorCode == RD_KAFKA_RESP_ERR_MEMBER_ID_REQUIRED))
+                        rd_kafka_cgrp_rebalance_revoke_all(
```

Comment:
yes. this is what i meant by the word "rebalance", and why i had previously named this `rd_kafka_cgrp_revoke_all_rebalance`, not `rd_kafka_cgrp_rebalance_revoke_all` as you preferred.

rather than add a comment, i'll change the name of the function to `rd_kafka_cgrp_revoke_all_rejoin`